compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -440,6 +441,8 @@ object desugar {
 
   /** The expansion of a class definition. See inline comments for what is involved */
   def classDef(cdef: TypeDef)(using Context): Tree = {
+    @threadUnsafe lazy val compilingScala2StdLib = ctx.settings.Yscala2Stdlib.value
```

Comment:
I don't think that will gain you anything to make it a lazy val, since that also adds inefficiency. If you want maximal performance, I'd do this:

```scala
  val caseClassInScala2StdLib = isClaseClass && ctx.settings.Yscala2Stdlib.value
```
after `isCaseCass` is defined. But it's probably overkill, and we can just test the settings repeatedly.